src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj

```diff
@@ -55,6 +56,7 @@
     <Compile Include="SerializeSqlTypesTest.cs" />
     <Compile Include="TestTdsServer.cs" />
     <Compile Include="SqlHelperTest.cs" />
+    <Compile Include="..\..\src\Microsoft\Data\Common\MultipartIdentifier.cs" />
```

Comment:
I understand why you did this to be able to access this source code here, but this approach will not give code coverage to netcore\M.D.S project csproj. This class will be executed from within this project.

You could instead provide access to test namespace by adding assembly tag:
`[assembly:InternalsVisibleTo("")]`

You'd not need ADB overrides too.

Reply:
In general I avoid InternalsVisibleTo because it's deprecated in netcore and wasn't a good idea to begin with. While it doesn't give code coverage it doesn't decrease code coverage either and it does increase test coverage so that the behaviour, which is most important, is now tested and deviations can be caught.

Reply:
Is there an official doc on it's deprecated status or why it's not a good idea? I mean there should be a better way of referencing internals if this is going to be deprecated! I don't think we need to limit ourselves.

If not, I'd rather increase code coverage than use this approach.

Reply:
Good question. I know there are many placed in corefx/runtime where it's been removed and in general they don't seem to want to use it but i don't have concrete information about why, some mention of build perf and trimming problems. I've asked on the discord if there's an official opinion on it's use.

There is the obvious negative that it adds the name of the test assemblies irrevocably into the released binary which is unneeded and since there is no strong name capability on core anything that wants to can see those names and pretend to be that assembly allowing them to screw with internals. 

I can understand why you'd want to increase coverage but contributors still can't see any of those coverage numbers and as I said I've increased test coverage., Requiring me to hit a target I can't evaluate seems unreasonable.

Reply:
It's actually a work in progress to be able to report coverage on PRs as they come along, there's just so much activity on higher priority. But we regularly run code coverage internally ([link here](https://dev.azure.com/sqlclientdrivers-ci/sqlclient/_build?definitionScope=%5CCoco)). I was saying in reference to new tests that we are adding, they should contribute to code coverage, and doesn't sound like we have large concerns over using the assembly tag.

> it adds the name of the test assemblies irrevocably into the released binary 
We already have a few assembly tags in other classes for internal testing, so it's not going to make additional difference.

Reply:
I've had a look and there are no test related InternalsVisibleTo attributes in the library. I also tried to look at the code coverage from your link but all I can find is an overall percentage and a .coverage file which I can't open.

Reply:
Sorry my bad. I think the alternate approach is using Reflection to access internal classes if not this way.
You're right, we should not use InternalsVisibleTo, as it causes assembly signing issues too.